packages/client/src/arpa_reporter/views/NewUpload.vue

```diff
@@ -1,111 +1,164 @@
+          validation: 'required',
+        },
+        {
+          type: 'select',
+          name: 'agencyId',
+          label: 'Agency Code',
+          options: this.agencySelectItems(),
+          validation: 'required',
+        },
+        {
+          type: 'text',
+          label: 'Expenditure Code',
+          name: 'expenditure_code',
+          validation: 'required',
+        },
+        {
+          type: 'file',
+          label: 'Workbook File',
+          name: 'spreadsheet',
+          validation: 'required',
+          uploader: this.uploadFile,
+          // Don't upload the file until the form is submitted
+          uploadBehavior: 'delayed',
+        },
+        {
+          type: 'textarea',
+          label: 'Notes',
+          name: 'notes',
+        },
+        {
+          component: 'div',
+          class: 'row',
+          children: [
+            {
+              name: 'submit',
+              type: 'submit',
+              label: 'Submit',
+              inputClass: ['btn', 'btn-primary'],
+            },
+            {
+              type: 'button',
+              label: 'Reset',
+              inputClass: ['btn', 'btn-danger', 'ml-2'],
+              '@click': this.onReset,
+            },
+          ],
+        },
+      ],
     };
   },
-  computed: {
-    uploadButtonLabel() {
-      return this.uploading ? 'Uploading...' : 'Upload';
-    },
-    uploadDisabled() {
-      return this.files === null || this.uploading;
-    },
+  created() {
+    this.values = this.initialValues();
   },
   methods: {
-    changeFiles(evt) {
-      this.files = evt.target.files;
+    reportingPeriodSelectItems() {
+      // create an object from this.$store.getters.viewableReportingPeriods where every key is period.id and every value is period.name
+      return Object.fromEntries(this.$store.getters.viewableReportingPeriods.map((period) => [period.id, period.name]));
     },
-    async onSubmit() {
-      const file = this.files[0];
-
-      if (!file) {
-        this.$refs.files.focus();
-        return;
-      }
-
-      this.error = null;
-      this.uploading = true;
-
-      const formData = new FormData();
-      formData.append('spreadsheet', file);
-
-      // TODO: (#896) Actually include data when submitting
-      // formData.append('notes', '<b>TEST & STRING!</b>');
-
+    agencySelectItems() {
+      return Object.fromEntries(this.$store.state.agencies.map((agency) => [agency.id, agency.name]));
+    },
+    // eslint-disable-next-line no-unused-vars
+    async uploadFile(file, progress, error, options) {
+      /* VueFormulate traditionally calls this method to upload the file
+         to some remote location, but for now, we don't want to do that.
+         It may eventually make sense to do that when we don't have access
+         to the file system, e.g in a lambda context. For now though,
+         we just return the file object. */
+      return file;
+    },
+    async onSubmit(formData) {
```

Comment:
```suggestion
    async onSubmit(data) {
```

Nit: just for clarity, could we give this a name that doesn't imply it is an instance of the `FormData` class?

Reply:
A good call, fixed